### ISSUE TITLE
Synchronization stuck in endless loop

### ISSUE THREAD

**Issue by BoboTiG** [dusk-network/rusk#1426]
**Describe the bug**

The node was syncing for hours, and in the morning when I checked the logs I saw it was stuck.

I can share full logs (200 Mb), and here is what seems to be interesting.

1) Block 65335 & 65336 were synced, then 65326 was received, and it failed with "invalid previous block hash":

```
2024-02-19T06:36:17.857756Z  INFO node::chain: event="block received" src="wire" blk_height=65335 blk_hash="60fd35e0b19b5d54...658ddc2922a6dfff"
2024-02-19T06:36:17.857764Z  INFO node::network: sending msg (GetData) to peer 78.47.49.3:9000
2024-02-19T06:36:17.857771Z  INFO node::chain: event="block received" src="wire" blk_height=65335 blk_hash="60fd35e0b19b5d54...658ddc2922a6dfff"
2024-02-19T06:36:17.857778Z  INFO node::network: sending msg (GetData) to peer 137.184.91.40:9000
2024-02-19T06:36:17.857889Z  INFO node::chain: event="block received" src="wire" blk_height=65336 blk_hash="7b27add83631896f...5dae78581ea68745"
2024-02-19T06:36:17.857897Z  INFO node::network: sending msg (GetData) to peer 109.199.97.102:9000
2024-02-19T06:36:17.857903Z  INFO node::chain: event="block received" src="wire" blk_height=65336 blk_hash="7b27add83631896f...5dae78581ea68745"
2024-02-19T06:36:17.857910Z  INFO node::network: sending msg (GetData) to peer 185.245.183.12:9000
2024-02-19T06:36:17.857915Z  INFO node::chain: event="block received" src="wire" blk_height=65336 blk_hash="7b27add83631896f...5dae78581ea68745"
2024-02-19T06:36:17.857922Z  INFO node::network: sending msg (GetData) to peer 158.220.96.81:9000
2024-02-19T06:36:17.857931Z  INFO node::chain: event="block received" src="wire" blk_height=65336 blk_hash="7b27add83631896f...5dae78581ea68745"
2024-02-19T06:36:17.857938Z  INFO node::network: sending msg (GetData) to peer 109.123.249.183:9000
2024-02-19T06:36:17.857950Z  INFO node::chain: event="block received" src="wire" blk_height=65336 blk_hash="7b27add83631896f...5dae78581ea68745"
2024-02-19T06:36:17.857957Z  INFO node::network: sending msg (GetData) to peer 5.78.98.17:9000
2024-02-19T06:36:17.857963Z  INFO node::chain: event="block received" src="wire" blk_height=65336 blk_hash="7b27add83631896f...5dae78581ea68745"
2024-02-19T06:36:17.857970Z  INFO node::network: sending msg (GetData) to peer 144.91.121.183:9000
2024-02-19T06:36:17.857978Z  INFO node::chain: event="block received" src="wire" blk_height=65336 blk_hash="7b27add83631896f...5dae78581ea68745"
2024-02-19T06:36:17.857985Z  INFO node::network: sending msg (GetData) to peer 159.89.127.99:9000
2024-02-19T06:36:17.857991Z  INFO node::chain: event="block received" src="wire" blk_height=65336 blk_hash="7b27add83631896f...5dae78581ea68745"
2024-02-19T06:36:17.857998Z  INFO node::network: sending msg (GetData) to peer 64.23.186.9:9000
2024-02-19T06:36:17.858004Z  INFO node::chain: event="block received" src="wire" blk_height=65336 blk_hash="7b27add83631896f...5dae78581ea68745"
2024-02-19T06:36:17.858011Z  INFO node::network: sending msg (GetData) to peer 159.65.40.207:9000
2024-02-19T06:36:17.858017Z  INFO node::chain: event="block received" src="wire" blk_height=65336 blk_hash="7b27add83631896f...5dae78581ea68745"
2024-02-19T06:36:17.858023Z  INFO node::network: sending msg (GetData) to peer 104.131.186.211:9000
2024-02-19T06:36:17.859473Z  INFO node::chain: event="block received" src="wire" blk_height=65326 blk_hash="b2425f7d3f7f3afc...61dad21994eb3592"
2024-02-19T06:36:17.861795Z ERROR node::chain: event="fsm::on_event failed" src="wire" err="invalid previous block hash"
```
Last 2 lines repeat indefinitely (until I saw the issue and stopped the node actually: so from 06:36:17 to 09:53:28).

2) I start the node again without doing anything else.

It seems it will start fresh from block 65322
```
2024-02-19T09:53:50.271035Z  INFO node::chain::acceptor: event="vm reverted" state_root="30016f8887cccbaa8ad999b01b89cbf79434623c8db46ff8da8b3b1b426594a6" is_final="true"
2024-02-19T09:53:50.271478Z  INFO node::chain::acceptor: event="block deleted" height=65325 iter=0 label="Attested" hash="737ed4770f6161e4d750b010449e28d01cc072ca7d8b05ed0d8aa2ea266132fb"
2024-02-19T09:53:50.271549Z  INFO node::chain::acceptor: event="block deleted" height=65324 iter=1 label="Attested" hash="43b9f081a169bfc497a3eb4a680595916ebe2e9da8763553887a5accfee7594f"
2024-02-19T09:53:50.271591Z  INFO node::chain::acceptor: event="block deleted" height=65323 iter=3 label="Attested" hash="188e4c4a3af46cd14ede29277e75ca12279040cd83f2f0c5c777f935b1341a7d"
2024-02-19T09:53:50.271700Z  INFO node::chain::acceptor: event="updating blockchain tip" height=65322 iter=0 state_root="30016f8887cccbaa8ad999b01b89cbf79434623c8db46ff8da8b3b1b426594a6"

# (...)

2024-02-19T09:53:58.902855Z  INFO node::chain: event="block received" src="wire" blk_height=65323 blk_hash="188e4c4a3af46cd1...c777f935b1341a7d"
2024-02-19T09:53:58.903150Z ERROR node::chain: event="fsm::on_event failed" src="wire" err="block already exists"
```

Last 2 lines are now filling logs indefinitely. Syncing seems in an endless loop.

---

**Platform**

 - Architecture: AMD64
 - OS: Linux Debian 12 (bookworm)

**Additional context**

there were no downtime, no internet loss, or anything that could disturb the sync.

**Comment by TMiNus** [comment]
Could you include the code you are running -  `rusk --version`

**Comment by BoboTiG** [comment]
I'm AFK right now, it is the version bundled with the ITN installer.

**Comment by TMiNus** [comment]
This issue may be fixed by https://github.com/dusk-network/rusk/pull/1420

**Comment by BoboTiG** [comment]
What is the best action to do now?

1. resync from the ground
2. wait for the new Rusk version + ITN installer

**Comment by BoboTiG** [comment]
I updated the node with the ITN installer v0.1.1, which brings `rusk 0.7.0 (ccba32c 2024-02-19)`: the node is still stuck into its endless loop.

**Comment by BoboTiG** [comment]
I upgraded manually `rusk` to benefit of latest commits, and the node is still stuck (on the bright side, there is less noise in the logs).

Here they are:
```
2024-02-19T17:49:40.488281Z  INFO node::chain::acceptor: event="VM state loaded" state_root="30016f8887cccbaa8ad999b01b89cbf79434623c8db46ff8da8b3b1b426594a6"
2024-02-19T17:49:40.489178Z  INFO node::chain::consensus: event="spawn consensus" id=1 round=65323 timeout={Proposal: 5s, Validation: 5s, Ratification: 5s} all=647 eligible=590
2024-02-19T17:49:40.489325Z  INFO node::chain: event="block received" src="wire" blk_height=68703 blk_hash="44265f4cb603a99d...9b6c5b9938daeb32"
2024-02-19T17:49:40.489376Z  INFO node::network: sending msg (GetData) to peer 212.24.108.115:9000
2024-02-19T17:49:40.489485Z  INFO node::chain: event="block received" src="wire" blk_height=68703 blk_hash="44265f4cb603a99d...9b6c5b9938daeb32"
2024-02-19T17:49:40.489498Z  INFO node::network: sending msg (GetData) to peer 64.23.133.158:9000
2024-02-19T17:49:40.490605Z  INFO node::chain: event="block received" src="wire" blk_height=68703 blk_hash="44265f4cb603a99d...9b6c5b9938daeb32"
2024-02-19T17:49:40.490635Z  INFO node::network: sending msg (GetData) to peer 24.199.111.175:9000
2024-02-19T17:49:40.491106Z  INFO node::chain: event="block received" src="wire" blk_height=68703 blk_hash="44265f4cb603a99d...9b6c5b9938daeb32"
2024-02-19T17:49:40.491132Z  INFO node::network: sending msg (GetData) to peer 207.180.243.37:9000
2024-02-19T17:49:40.504534Z  INFO node::chain: event="block received" src="wire" blk_height=65323 blk_hash="188e4c4a3af46cd1...c777f935b1341a7d"
2024-02-19T17:49:40.507650Z ERROR node::chain: event="fsm::on_event failed" src="wire" err="block already exists"
2024-02-19T17:49:40.518875Z  INFO node::chain: event="block received" src="wire" blk_height=65323 blk_hash="188e4c4a3af46cd1...c777f935b1341a7d"
2024-02-19T17:49:40.519268Z ERROR node::chain: event="fsm::on_event failed" src="wire" err="block already exists"
2024-02-19T17:49:40.638262Z  INFO node::chain: event="block received" src="wire" blk_height=65323 blk_hash="188e4c4a3af46cd1...c777f935b1341a7d"
2024-02-19T17:49:40.638668Z ERROR node::chain: event="fsm::on_event failed" src="wire" err="block already exists"
2024-02-19T17:49:40.640134Z  INFO node::chain: event="block received" src="wire" blk_height=65323 blk_hash="188e4c4a3af46cd1...c777f935b1341a7d"
2024-02-19T17:49:40.640308Z ERROR node::chain: event="fsm::on_event failed" src="wire" err="block already exists"
2024-02-19T17:49:45.491455Z  INFO main{round=65323 iter=0 name=Proposal pk="mvF5VaJPCjyUN1tu"}: dusk_consensus::execution_ctx: event="timeout-ed"
2024-02-19T17:49:49.106643Z  WARN node::databroker: error on handling msg: could not find locator block
2024-02-19T17:49:50.496624Z  INFO main{round=65323 iter=0 name=Validation pk="mvF5VaJPCjyUN1tu"}: dusk_consensus::execution_ctx: event="timeout-ed"
2024-02-19T17:49:51.607813Z  INFO node::chain: event="block received" src="wire" blk_height=68704 blk_hash="9284f322daf7230b...8afb4766d151bdb5"
2024-02-19T17:49:51.608019Z  INFO node::network: sending msg (GetData) to peer 64.23.133.158:9000
2024-02-19T17:49:51.758111Z  INFO node::chain: event="block received" src="wire" blk_height=65323 blk_hash="188e4c4a3af46cd1...c777f935b1341a7d"
2024-02-19T17:49:51.758410Z ERROR node::chain: event="fsm::on_event failed" src="wire" err="block already exists"
2024-02-19T17:49:51.980972Z  INFO node::chain: event="block received" src="wire" blk_height=68704 blk_hash="9284f322daf7230b...8afb4766d151bdb5"
2024-02-19T17:49:51.981242Z  INFO node::network: sending msg (GetData) to peer 64.23.186.9:9000
2024-02-19T17:49:52.129722Z  INFO node::chain: event="block received" src="wire" blk_height=65323 blk_hash="188e4c4a3af46cd1...c777f935b1341a7d"
2024-02-19T17:49:52.129989Z ERROR node::chain: event="fsm::on_event failed" src="wire" err="block already exists"
2024-02-19T17:49:52.215020Z  INFO node::chain: event="block received" src="wire" blk_height=68704 blk_hash="9284f322daf7230b...8afb4766d151bdb5"
2024-02-19T17:49:52.215160Z  INFO node::network: sending msg (GetData) to peer 117.7.117.120:9000
2024-02-19T17:49:52.443498Z  INFO node::chain: event="block received" src="wire" blk_height=68704 blk_hash="9284f322daf7230b...8afb4766d151bdb5"
2024-02-19T17:49:52.443726Z  INFO node::network: sending msg (GetData) to peer 185.92.1.19:9000
2024-02-19T17:49:52.491613Z  INFO node::chain: event="block received" src="wire" blk_height=65323 blk_hash="188e4c4a3af46cd1...c777f935b1341a7d"
2024-02-19T17:49:52.491940Z ERROR node::chain: event="fsm::on_event failed" src="wire" err="block already exists"
2024-02-19T17:49:52.499790Z  INFO node::chain: event="block received" src="wire" blk_height=65323 blk_hash="188e4c4a3af46cd1...c777f935b1341a7d"
2024-02-19T17:49:52.500121Z ERROR node::chain: event="fsm::on_event failed" src="wire" err="block already exists"
2024-02-19T17:49:54.021820Z  INFO node::chain: event="block received" src="wire" blk_height=68704 blk_hash="9284f322daf7230b...8afb4766d151bdb5"
2024-02-19T17:49:54.021968Z  INFO node::network: sending msg (GetData) to peer 68.183.28.227:9000
2024-02-19T17:49:54.112713Z  INFO node::chain: event="block received" src="wire" blk_height=65323 blk_hash="188e4c4a3af46cd1...c777f935b1341a7d"
2024-02-19T17:49:54.112928Z ERROR node::chain: event="fsm::on_event failed" src="wire" err="block already exists"
2024-02-19T17:49:55.503465Z  INFO main{round=65323 iter=0 name=Ratification pk="mvF5VaJPCjyUN1tu"}: dusk_consensus::execution_ctx: event="timeout-ed"
2024-02-19T17:49:56.702203Z  INFO node::chain: event="block received" src="wire" blk_height=68704 blk_hash="9284f322daf7230b...8afb4766d151bdb5"
2024-02-19T17:49:56.702329Z  INFO node::network: sending msg (GetData) to peer 137.184.91.40:9000
2024-02-19T17:49:56.852073Z  INFO node::chain: event="block received" src="wire" blk_height=65323 blk_hash="188e4c4a3af46cd1...c777f935b1341a7d"
2024-02-19T17:49:56.852288Z ERROR node::chain: event="fsm::on_event failed" src="wire" err="block already exists"
2024-02-19T17:49:57.445465Z  INFO node::chain: event="block received" src="wire" blk_height=68704 blk_hash="9284f322daf7230b...8afb4766d151bdb5"
2024-02-19T17:49:57.445566Z  INFO node::network: sending msg (GetData) to peer 103.150.191.223:9000
2024-02-19T17:49:57.622243Z  INFO node::chain: event="block received" src="wire" blk_height=65323 blk_hash="188e4c4a3af46cd1...c777f935b1341a7d"
2024-02-19T17:49:57.622435Z ERROR node::chain: event="fsm::on_event failed" src="wire" err="block already exists"
2024-02-19T17:49:59.518247Z  INFO rusk::chain::vm: Received preverify request
2024-02-19T17:49:59.572527Z  INFO node::mempool: event="transaction accepted" hash="76cdec18a912ffa065061f7b3fbba96efb19c47a93d612a122ec0fb7cfd13d0f"
2024-02-19T17:50:02.271792Z  INFO kadcast::maintainer: TableMaintainer::monitor_buckets back to sleep
2024-02-19T17:50:02.505787Z  INFO main{round=65323 iter=1 name=Proposal pk="mvF5VaJPCjyUN1tu"}: dusk_consensus::execution_ctx: event="timeout-ed"
2024-02-19T17:50:09.510514Z  INFO main{round=65323 iter=1 name=Validation pk="mvF5VaJPCjyUN1tu"}: dusk_consensus::execution_ctx: event="timeout-ed"
2024-02-19T17:50:16.513430Z  INFO main{round=65323 iter=1 name=Ratification pk="mvF5VaJPCjyUN1tu"}: dusk_consensus::execution_ctx: event="timeout-ed"
2024-02-19T17:50:17.112393Z  WARN node::databroker: error on handling msg: could not find locator block
2024-02-19T17:50:17.276960Z  INFO node::network: sending msg (GetInv) to peer 45.67.221.254:9000
2024-02-19T17:50:17.446444Z  INFO node::chain::fsm: event="fsm::idle" height=65322 iter=0 timeout_sec=20 finalized_height=65322
2024-02-19T17:50:25.560997Z  INFO main{round=65323 iter=2 name=Proposal pk="mvF5VaJPCjyUN1tu"}: dusk_consensus::execution_ctx: event="timeout-ed"
2024-02-19T17:50:34.580396Z  INFO main{round=65323 iter=2 name=Validation pk="mvF5VaJPCjyUN1tu"}: dusk_consensus::execution_ctx: event="timeout-ed"
2024-02-19T17:50:37.448762Z  INFO node::chain::fsm: event="fsm::idle" height=65322 iter=0 timeout_sec=20 finalized_height=65322
2024-02-19T17:50:43.265603Z  INFO node::chain: event="block received" src="wire" blk_height=68705 blk_hash="dfd59a4e9edcd4d5...444e72d890dfaaab"
2024-02-19T17:50:43.265697Z  INFO node::network: sending msg (GetData) to peer 65.109.229.92:9000
2024-02-19T17:50:43.297286Z  INFO node::chain: event="block received" src="wire" blk_height=65323 blk_hash="188e4c4a3af46cd1...c777f935b1341a7d"
2024-02-19T17:50:43.297550Z ERROR node::chain: event="fsm::on_event failed" src="wire" err="block already exists"
2024-02-19T17:50:43.583156Z  INFO main{round=65323 iter=2 name=Ratification pk="mvF5VaJPCjyUN1tu"}: dusk_consensus::execution_ctx: event="timeout-ed"
2024-02-19T17:50:43.740608Z  INFO node::chain: event="block received" src="wire" blk_height=68705 blk_hash="dfd59a4e9edcd4d5...444e72d890dfaaab"
2024-02-19T17:50:43.740694Z  INFO node::network: sending msg (GetData) to peer 94.103.83.160:9000
2024-02-19T17:50:43.783037Z  INFO node::chain: event="block received" src="wire" blk_height=65323 blk_hash="188e4c4a3af46cd1...c777f935b1341a7d"
2024-02-19T17:50:43.784878Z ERROR node::chain: event="fsm::on_event failed" src="wire" err="block already exists"
2024-02-19T17:50:54.584790Z  INFO main{round=65323 iter=3 name=Proposal pk="mvF5VaJPCjyUN1tu"}: dusk_consensus::execution_ctx: event="timeout-ed"
2024-02-19T17:50:59.530308Z  INFO node::chain: event="block received" src="wire" blk_height=68706 blk_hash="14a07e051768e5ee...322a1bb603396c0b"
2024-02-19T17:50:59.530403Z  INFO node::network: sending msg (GetData) to peer 68.183.28.227:9000
2024-02-19T17:50:59.622993Z  INFO node::chain: event="block received" src="wire" blk_height=65323 blk_hash="188e4c4a3af46cd1...c777f935b1341a7d"
2024-02-19T17:50:59.625813Z ERROR node::chain: event="fsm::on_event failed" src="wire" err="block already exists"
```

`rusk` version from commit 0a73f3a7:
```console
$ rusk --version
rusk 0.7.0 (0a73f3a7 2024-02-19)
```

**Comment by BoboTiG** [comment]
I'll keep my node stuck in the case you would like me to try commands or a custom `rusk` version in order to fix it. LMK if that would be helpful.

**Comment by herr-seppia** [comment]
Yes, that's the culprit
```
2024-02-19T09:53:58.903150Z ERROR node::chain: event="fsm::on_event failed" src="wire" err="block already exists"
```

There was a bug in how the fallback procedure were deleting blocks, that we fixed with latest 0.1.1 installer.
Unluckily your node restarted before applying that patch, so the only way to get unstucked (in your case) is to reset from scratch
